### PR TITLE
AITriggers support upgrades

### PR DESCRIPTION
**Pull request by Thrifinesma**
In vanilla, it is not allowed that use Building Upgrades in AITriggerTypes. This would fix that.

### CODE REVIEW

File: src/Misc/Hooks.BugFixes.cpp [hunk 1]
```diff
@@ -293,4 +294,40 @@ DEFINE_HOOK(0x51BB6E, TechnoClass_AI_TemporalTargetingMe_Fix, 0x6) // InfantryCl
...
+		if (pPlugin && pHouse->OwnedBuildingTypes1.GetItemCount(pPlugin->ArrayIndex) > 0)
+		{
+			for (auto const& pBld : pHouse->Buildings)
+			{
+				if (pBld->Type != pPlugin)
+					continue;
+
+				for (const auto& pUpgrade : pBld->Upgrades)
+					if (pUpgrade == pType)
+						count++;
```
Comment (YRIDZE) [inline src/Misc/Hooks.BugFixes.cpp:322]:
```suggestion
				for (const auto& pUpgrade : pBld->Upgrades)
				{
					if (pUpgrade == pType)
						count++;
				}
```

File: src/Misc/Hooks.BugFixes.cpp [hunk 2]
```diff
@@ -293,4 +294,40 @@ DEFINE_HOOK(0x51BB6E, TechnoClass_AI_TemporalTargetingMe_Fix, 0x6) // InfantryCl
...
+			}
+		}
+	}
+	else
+	{
+		count = pHouse->OwnedBuildingTypes1.GetItemCount(idxBld);
+	}
+
+	R->EAX(count);
+	return R->Origin() + 0xC;
```
Comment (YRIDZE) [inline src/Misc/Hooks.BugFixes.cpp:332]:
```suggestion
	R->EAX(count);
	
	return R->Origin() + 0xC;
```

File: src/Misc/Hooks.BugFixes.cpp [hunk 3]
```diff
@@ -293,4 +294,40 @@ DEFINE_HOOK(0x51BB6E, TechnoClass_AI_TemporalTargetingMe_Fix, 0x6) // InfantryCl
...
+// Author: Uranusian
+DEFINE_HOOK_AGAIN(0x41EEE3, AITriggerTypeClass_Condition_SupportPowersup, 0x7)	//AITriggerTypeClass_OwnerHouseOwns_SupportPowersup
+DEFINE_HOOK(0x41EB43, AITriggerTypeClass_Condition_SupportPowersup, 0x7)		//AITriggerTypeClass_EnemyHouseOwns_SupportPowersup
+{
+	GET(HouseClass*, pHouse, EDX);
+	GET(int, idxBld, EBP);
+	auto const pType = BuildingTypeClass::Array->Items[idxBld];
+	auto const pPowerup = pType->PowersUpBuilding;
+	int count = 0;
+	// TODO: support Phobos extended Building Upgrades logic
```
Comment (YRIDZE) [inline src/Misc/Hooks.BugFixes.cpp:309]:
```suggestion
	
	// TODO: support Phobos extended Building Upgrades logic
```

**Review by YRIDZE — CHANGES_REQUESTED** [review]
Code looks pretty good. Please, fix some codestyle mismatches.

**Comment by ayylmaoRotE** [comment]
tested and can confirm working

**Review by Metadorius — APPROVED** [review]
image

**Review by Metadorius — APPROVED** [review]

**Pull request merged**